src/doc-printer.js

```diff
@@ -3,6 +3,33 @@
 const MODE_BREAK = 1;
 const MODE_FLAT = 2;
 
+function rootIndent() {
+  return {
+    indent: 0,
+    align: {
+      spaces: 0,
+      tabs: 0
+    }
+  };
+}
+
+function indent(ind) {
+  return {
+    indent: ind.indent + 1,
+    align: ind.align
+  };
+}
+
+function align(ind, n) {
+  return {
+    indent: ind.indent,
+    align: {
+      spaces: ind.align.spaces + n,
+      tabs: ind.align.tabs + (n ? 1 : 0)
```

Comment:
I'm especially curious if you have tab width = 3.

The inner one is
```
<indent><indent><2 space><indent><2 space><indent>
```

If you do tabWidth = 3 then you have

```
tab(3) tab(3) space(2) tab(3) space(2) tab(3)
```

if you output it like this, then the tab after the spaces is going to be width = 1 in order to align, which is incorrect.

What you are doing instead is

```
tab(3) tab(3) tab(3) space(2) space(2)
```

which seems to be correct!